internal/mesh/internal/controllers/xds/controller_test.go

```diff
@@ -112,558 +115,613 @@ func mockFetcher() (*pbproxystate.TrustBundle, error) {
 
 // This test ensures when a ProxyState is deleted, it is no longer tracked in the mappers.
 func (suite *xdsControllerTestSuite) TestReconcile_NoProxyStateTemplate() {
-	// Track the id of a non-existent ProxyStateTemplate.
-	proxyStateTemplateId := resourcetest.Resource(pbmesh.ProxyStateTemplateType, "not-found").ID()
-	suite.mapper.TrackItem(proxyStateTemplateId, []resource.ReferenceOrID{})
-	suite.leafMapper.TrackItem(proxyStateTemplateId, []resource.ReferenceOrID{})
-	require.False(suite.T(), suite.mapper.IsEmpty())
-	require.False(suite.T(), suite.leafMapper.IsEmpty())
-
-	// Run the reconcile, and since no ProxyStateTemplate is stored, this simulates a deletion.
-	err := suite.ctl.Reconcile(context.Background(), suite.runtime, controller.Request{
-		ID: proxyStateTemplateId,
-	})
-	require.NoError(suite.T(), err)
+	suite.runTestCaseWithTenancies(func(tenancy *pbresource.Tenancy) {
+		// Track the id of a non-existent ProxyStateTemplate.
+		proxyStateTemplateId := resourcetest.Resource(pbmesh.ProxyStateTemplateType, "not-found").WithTenancy(tenancy).ID()
+
+		defer suite.client.MustDelete(suite.T(), proxyStateTemplateId)
```

Comment:
`t.Cleanup` is slightly better in tests because it handles panics, parallel tests, etc.